shared/src/commonMain/kotlin/com/example/tasky/auth/domain/ISessionManager.kt

```diff
@@ -0,0 +1,21 @@
+package com.example.tasky.auth.domain
+
+import com.example.tasky.auth.domain.model.RefreshToken
+import io.ktor.client.plugins.auth.providers.BearerTokens
+
+interface ISessionManager {
+    suspend fun loadTokens(): BearerTokens
+
+    suspend fun loadAccessTokenBody(): RefreshToken?
+
+    suspend fun updateAccessToken(
+        newToken: String,
+        expirationTimestamp: Long,
+    )
+
+    suspend fun removeToken()
+
+    suspend fun getFullName(): String?
+
+    suspend fun getUserId(): String?
```

Comment:
minor - how about encapsulating this into a single data class? I called it `AuthInfo` which contains all data related to an authenticated session:

```
data class AuthInfo(
  val accessToken: String,
  val refreshToken: String,
  val userId: String,
  val fullName: String,
  val email: String
)
```

I found it easy to work with in cases where I needed multiple pieces of data at the same time. After a single null-check, I have all the data I need. Saving and retrieving the data is done via two. functions:

```
interface SessionStorage {
  suspend fun getSession(): AuthInfo?
  suspend fun setSession(info: AuthInfo?)
}
```